### PR TITLE
Add setting for publishing on Sonatype

### PR DESCRIPTION
**Pull request by rafaparadela**
This PR, not only adds settings to configure publishing snapshots but also it helped me to understand how it works. Fancy documentation is almost ready.

@FPerezP @fedefernandez @javipacheco Could you review please? Thank you in advance.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -2,7 +2,7 @@ organization := "com.fortysevendeg"
 
 name := "github4s"
 
-version := "0.1"
+version := "0.1-SNAPSHOT"
```
Comment (javipacheco) [inline build.sbt]: We should create  `0.1` version and later we can create a new PR with `0.1-SNAPSHOT`

File: publish.sbt [hunk 2]
```diff
@@ -0,0 +1,54 @@
+organizationName := "47 Degrees"
+
+organizationHomepage := Some(new URL("http://47deg.com"))
+
+publishMavenStyle := true
+
+startYear := Some(2015)
```
Comment (javipacheco) [inline publish.sbt]: 2016?

File: build.sbt (hunk 1 above)
Comment (rafaparadela) [inline build.sbt]: Makes sense

File: publish.sbt (hunk 2 above)
Comment (rafaparadela) [inline publish.sbt]: Done ;-)

**Comment by javipacheco** [comment]
LGTM!

**Pull request merged**